tests/phpunit/tests/block-template-utils.php

```diff
@@ -471,4 +473,79 @@ public function test_inject_ignored_hooked_blocks_metadata_attributes_into_templ
 			'The hooked block was not injected into the anchor block\'s ignoredHookedBlocks metadata.'
 		);
 	}
+
+	/**
+	 * @ticket 60754
+	 *
+	 * @covers inject_ignored_hooked_blocks_metadata_attributes
+	 */
+	public function test_inject_ignored_hooked_blocks_metadata_attributes_applies_filter_correctly() {
+		global $wp_current_filter;
+		// Mock currently set filter.
+		// The $wp_current_filter global is reset during teardown by the unit test base class.
+		$wp_current_filter[] = 'rest_pre_insert_wp_template';
+
+		$action = new MockAction();
+		add_filter( 'hooked_block_types', array( $action, 'filter' ), 10, 4 );
+
+		$id      = self::TEST_THEME . '//' . 'my_template';
+		$request = new WP_REST_Request( 'POST', '/wp/v2/templates/' . $id );
+		$request->set_param( 'id', $id );
+
+		$changes                = new stdClass();
+		$changes->ID            = self::$template_post->ID;
+		$changes->post_type     = 'wp_template';
+		$changes->post_author   = 123;
+		$changes->post_name     = 'my-updated-template';
+		$changes->post_title    = 'My updated Template';
+		$changes->post_content  = '<!-- wp:tests/anchor-block -->Hello<!-- /wp:tests/anchor-block -->';
+		$changes->post_excerpt  = 'Displays a single post on your website unless a custom template...';
+		$changes->post_status   = 'publish';
+		$changes->post_modified = '2021-07-01 12:00:00'; // FIXME
+		$changes->meta_input    = array(
+			'origin'           => 'custom',
+			'is_wp_suggestion' => true,
+		);
+		$changes->tax_input     = array(
+			'wp_theme' => self::TEST_THEME,
+		);
+
+		inject_ignored_hooked_blocks_metadata_attributes( $changes, $request );
+
+		$args              = $action->get_args();
+		$anchor_block_type = end( $args )[2];
+		$context           = end( $args )[3];
+
+		$this->assertSame( 'tests/anchor-block', $anchor_block_type );
+		$this->assertInstanceOf( 'WP_Block_Template', $context );
+
+		$post_to_template_key_map = array(
+			'ID'            => 'wp_id',
+			'post_author'   => 'author',
+			'post_name'     => 'slug',
+			'post_content'  => 'content',
+			'post_title'    => 'title',
+			'post_excerpt'  => 'description',
+			'post_type'     => 'type',
+			'post_status'   => 'status',
+			'post_modified' => 'modified',
+		);
+
+		$expected = array();
+		foreach ( $post_to_template_key_map as $post_key => $template_key ) {
+			if ( isset( $changes->$post_key ) ) {
+				$expected[ $template_key ] = $changes->$post_key;
+			}
+		}
+		$expected['id']             = self::TEST_THEME . '//' . 'my-updated-template';
+		$expected['origin']         = 'custom';
+		$expected['source']         = 'custom';
+		$expected['theme']          = self::TEST_THEME;
+		$expected['is_custom']      = false;
+		$expected['has_theme_file'] = false;
+		$expected['post_types']     = null;
+		$expected['area']           = null;
+
+		$this->assertEquals( $expected, (array) $context );
```

Comment:
Let's turn these into individual assertions, as we used to have before https://github.com/WordPress/wordpress-develop/commit/ce3c67bcaa31ae0c7288f235a24e55b782ed1ac4.